classes/module/Module.php

```diff
@@ -1982,28 +1982,45 @@ public function l($string, $specific = false, $locale = null)
     public function updatePosition($id_hook, $way, $position = null)
     {
         foreach (Shop::getContextListShopID() as $shop_id) {
-            $sql = 'SELECT hm.`id_module`, hm.`position`, hm.`id_hook`
+            $sqlQueryToGetAvailableHookPositions = 'SELECT hm.`id_module`, hm.`position`, hm.`id_hook`
                     FROM `' . _DB_PREFIX_ . 'hook_module` hm
                     WHERE hm.`id_hook` = ' . (int) $id_hook . ' AND hm.`id_shop` = ' . $shop_id . '
                     ORDER BY hm.`position` ' . ($way ? 'ASC' : 'DESC');
-            if (!$res = Db::getInstance()->executeS($sql)) {
+
+            if (!$sqlResult = Db::getInstance()->executeS($sqlQueryToGetAvailableHookPositions)) {
+                // no hook positions available
                 continue;
             }
+            if (count($sqlResult) === 1) {
+                // if there is only 1 position available, it cannot be updated
+                return false;
+            }
 
-            foreach ($res as $key => $values) {
-                if ((int) $values[$this->identifier] == (int) $this->id) {
-                    $k = $key;
+            foreach ($sqlResult as $positionNumber => $positionSettings) {
+                $thisIsTheSettingsForThisModule = ((int) $positionSettings[$this->identifier] == (int) $this->id);
+
+                if ($thisIsTheSettingsForThisModule) {
+                    $thisModulePositionNumber = $positionNumber;
 
                     break;
                 }
             }
 
-            if (!isset($k) || !isset($res[$k]) || !isset($res[$k + 1])) {
+            if (!isset($thisModulePositionNumber)) {
+                // could not find hook positions for this module
+                return false;
+            }
+            if (!isset($sqlResult[$thisModulePositionNumber])) {
+                // ok this one is really weird
```

Comment:
🤣

Reply:
Don't ask why.. We don't know :sob:

Reply:
I kept it because it was the legacy behavior but this case is logically impossible.